api/src/main/java/org/bahmni/module/events/api/listener/AppointmentAdvice.java

```diff
@@ -0,0 +1,61 @@
+package org.bahmni.module.events.api.listener;
+
+import org.bahmni.module.events.api.model.BahmniEventType;
+import org.bahmni.module.events.api.model.Event;
+import org.openmrs.module.appointments.model.Appointment;
+import org.openmrs.module.appointments.web.mapper.AppointmentMapper;
+import org.springframework.aop.AfterReturningAdvice;
+import org.springframework.aop.MethodBeforeAdvice;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.ApplicationEventPublisherAware;
+import org.springframework.lang.NonNull;
+
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import static org.bahmni.module.events.api.model.BahmniEventType.*;
+
+public class AppointmentAdvice implements AfterReturningAdvice, ApplicationEventPublisherAware, MethodBeforeAdvice {
+
+	private ApplicationEventPublisher eventPublisher;
+
+	private final ThreadLocal<Map<String,Integer>> threadLocal = new ThreadLocal<>();
+	private final String APPOINTMENT_ID_KEY = "appointmentId";
+
+	private final ApplicationContext applicationContext;
+
+	public AppointmentAdvice(ApplicationContext applicationContext) {
+		this.applicationContext = applicationContext;
+	}
+
+	@Override
+	public void afterReturning(Object returnValue, Method method, Object[] arguments, Object target) {
+		Map<String, Integer> appointmentInfo = threadLocal.get();
+		if (appointmentInfo != null) {
```

Comment:
Add TODO to signify that this condition is the workaround so that we don't publish duplicate events.